apps/libs/sdk/src/laddercaster/program/CasterContext.ts

```diff
@@ -810,89 +758,34 @@ export class CasterContext {
   //HELPERS
   private async getGameTurnData(game: Game, gameAccount: PublicKey, turn = -1) {
     const turnNumber = turn !== -1 ? turn : game?.turnInfo?.turn;
-
-    return await anchor.web3.PublicKey.findProgramAddress(
-      [
-        Buffer.from('turn_data'),
-        gameAccount.toBuffer(),
-        Buffer.from(anchor.utils.bytes.utf8.encode(String(turnNumber))),
-      ],
-      this.client.program.programId,
-    );
+    if(turn!=-1 && turn!=game?.turnInfo?.turn){
+      return await anchor.web3.PublicKey.findProgramAddress(
+        [
+          Buffer.from('turn_data'),
+          gameAccount.toBuffer(),
+          Buffer.from(anchor.utils.bytes.utf8.encode(String(turnNumber))),
+        ],
+        gameConstantsContext.Client.program.programId,
+      );
+    }
+   return [gameConstantsContext.turnData,123]
```

Comment:
any reason why its returning 123?

Reply:
Ah that was the nonce/bump it returns from the findProgramAddress,  but forgot to just destructure. Fixed it now :)